### ISSUE TITLE
Compilation fails on empty values on v-bind

### ISSUE THREAD

**Issue by stephan-v** [vuejs/vue#7973]
### Version
2.5.13

### Reproduction link
[https://jsfiddle.net/jt3hjuwg/](https://jsfiddle.net/jt3hjuwg/)

### Steps to reproduce
I am referencing this closed issue:

https://github.com/vuejs/vue/issues/3969

I feel like something should be done about this issue because it has left me with some white pages along the way and without a clear error message. If you pass in a null value as a prop the entire compilation breaks down with your evaluated html simply being dumped into the console.

Is there really not a better way to handle this? Obviously a user should or could check for these values before actually showing the component but in large application a piece of data can sometimes be null either by accident or oversight

If nothing can be done about this could someone at least please explain why this happens because in my eyes a component not getting a prop does not justify the entire compilation failing.

### What is expected?
The application should not fail compilation.

### What is actually happening?
It fails compilation.

**Comment by posva** [comment]
Maybe a warning can be added in dev mode

**Comment by stephan-v** [comment]
@kidplug 

In the example given you are correct, but what if I were to pass along a null value from something like PHP though? It still fails, it acts as If I have not given it anything at all and the entire compilation breaks.

```
<?php $nullValue = null; ?>

<review :content="<?= $nullValue ?>"></review>

```

**Comment by jacekkarczmarczyk** [comment]
You should do
```
<?php $value = whatever; ?>

<review :content="<?= json_encode($value) ?>"></review>
```

**Comment by yyx990803** [comment]
Note this is supposed to fail, because by definition, using `v-bind` means the content is treated as JavaScript expression, so it's the equivalent of writing the following in JavaScript:

``` js
createElement('review', { attrs: { content: }})
```

What we can do is give a proper compilation warning instead of a runtime error.

**Comment by astronomersiva** [comment]
I am interested in taking this up :) It would be really helpful if someone guides me with where to start off.

**Comment by lyhper** [comment]
I'd like to have a try

**Comment by leevigraham** [comment]
Related: https://github.com/vuejs/vue/issues/3969

**Comment by stephan-v** [comment]
By the way what is being suggested here by @jacekkarczmarczyk will not work for numbers since they are being converted to strings this way.

```
<?php $value = whatever; ?>

<review :content="<?= json_encode($value) ?>"></review>
```

**Comment by jacekkarczmarczyk** [comment]
No, numbers will be numbers, if you have strings it means than you have strings instead of numbers
![image](https://user-images.githubusercontent.com/15625235/40982918-a8eef3be-6908-11e8-88af-e96e04567482.png)

**Comment by yyx990803** [comment]
closed via #7940
